File: Data Structures & Algorithms/insert-new-interval/submission-1.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        
        if not intervals:
            return [newInterval]
        if not newInterval:
            return [intervals]

        res = []
        i = 0
        flag = True
        for interval in intervals:
            if interval[1] < newInterval[0]:
                res.append(interval)
            
            elif interval[0]> newInterval[1]:
                if flag:
                    res.append(newInterval)
                    flag = False
                res.append(interval)
            else:
                newInterval[0] = min(interval[0],newInterval[0])
                newInterval[1] = max(interval[1],newInterval[1])
        
        if flag:
            res.append(newInterval)
        
        return res
        

File: Data Structures & Algorithms/meeting-schedule-ii/submission-1.cpp
/**
 * Definition of Interval:
 * class Interval {
 * public:
 *     int start, end;
 *     Interval(int start, int end) {
 *         this->start = start;
 *         this->end = end;
 *     }
 * }
 */

class Solution {
public:
    int minMeetingRooms(vector<Interval>& intervals) {
        vector<pair<int, int>> time;
        for (const auto& i : intervals) {
            time.push_back({i.start, 1});
            time.push_back({i.end, -1});
        }

        sort(time.begin(), time.end(), [](auto& a, auto& b) {
            return a.first == b.first ? a.second < b.second : a.first < b.first;
        });

        int res = 0, count = 0;
        for (const auto& t : time) {
            count += t.second;
            res = max(res, count);
        }
        return res;
    }
};